app/assets/javascripts/client/common_create/CreateService.js

```diff
@@ -80,5 +85,45 @@
             service.emitError('Could not save assessment');
           });
     };
+
+    this.assignAndSaveInventory = function(inventory) {
+      if (inventory.message === null || inventory.message === '') {
+        this.alertError = true;
+        return;
+      }
+
+      if ($window.confirm('Are you sure you want to start the inventory and invite all your participants?')) {
+        this.alertError = false;
+        this.saveInventory(inventory, true)
+            .then(function() {
+              $location.path('/inventories');
+            });
+      }
+    };
+
+    this.saveInventory = function(inventory, assign) {
+      if (inventory.name === '') {
+        this.emitError('Inventory needs a name!');
+        return;
+      }
+
+      inventory.district_id = this.district.id;
+      service.toggleSavingState();
```

Comment:
Would it be more consistent to either use `this` or, preferably, `service` in all instances?